server/channels/app/audit.go

```diff
@@ -156,3 +162,66 @@ func (s *Server) onAuditTargetQueueFull(qname string, maxQSize int) bool {
 func (s *Server) onAuditError(err error) {
 	s.Log().Error("Audit Error", mlog.Err(err))
 }
+
+func (a *App) AddAuditLogCertificate(rctx request.CTX, fileData *multipart.FileHeader) *model.AppError {
+	file, err := fileData.Open()
+	if err != nil {
+		return model.NewAppError("AddSamlCertificate", "api.admin.add_certificate.open.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
+	}
+	defer file.Close()
+
+	data, err := io.ReadAll(file)
+	if err != nil {
+		return model.NewAppError("AddSamlCertificate", "api.admin.add_certificate.saving.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
+	}
+
+	err = a.Srv().platform.SetConfigFile(AuditCertificateFilename, data)
+	if err != nil {
+		return model.NewAppError("AddSamlCertificate", "api.admin.add_certificate.saving.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
+	}
+
+	cfg := a.Config().Clone()
+
+	*cfg.ExperimentalAuditSettings.Certificate = AuditCertificateFilename
```

Comment:
I reviewed our docs on this config and I am not exactly sure how `Certificate` interacts with the `AdvancedLoggingJSON `. Did you happen to come across what this value does when `AdvancedLoggingJSON` references a cert file?

Reply:
The `Certificate` config is basically just facilitating the re-use of system console components. Otherwise I'd have had to do a fully custom page, and add a bunch of logic to Provisioner/CWS/Enterprise for listing/fetching certificates and such (only create/update/delete exist in provisioner right now). The advancedloggingjson should be using the mounted version from the provisioner, placed at either `/mattermost/custom-certs` or a different path from `AUDIT_LOGGING_CERT_MOUNT_PATH` env variable

Reply:
Okay makes sense. Thanks!